### PR TITLE
Add tests for sub_group_mask member functions insert_bits() and extract_bits()

### PR DESCRIPTION
**Pull request by vasilytric**
`sub_group_mask` it's an extension, documentations can be found here: [intel/llvm/doc/SubGroupMask](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/SubGroupMask/SubGroupMask.asciidoc)

This PR depends on https://github.com/KhronosGroup/SYCL-CTS/pull/200
`sub_group_mask_common.h` file can be found in PR#200

### CODE REVIEW

File: tests/common/type_coverage.h [hunk 1]
```diff
@@ -231,4 +230,49 @@ void for_all_types_vectors_marray(const named_type_pack<types...>& typeList,
...
+   */
+  size_t typeNameIndex = 0;
+
+  int packExpansion[] = {(
+      for_type_and_marrays<action, types, actionArgsT...>(
+          std::forward<argsT>(args)..., typeList.names[typeNameIndex]),
+      ++typeNameIndex,
+      0  // Dummy initialization value
+      )...};
+  static_cast<void>(packExpansion);
```
Comment (keryell) [inline tests/common/type_coverage.h:276]: Just back from CppCon, it looks like I have seen simpler way like:
```c++
#include <iostream>
void print(auto a, int i) {
    std::cout << i << ':' << a << std::endl;
}
void expand(auto... a) {
    int i = 0;
    ((print(a, i), i++),...);
}

int main() {
    expand(1,2,3);
}
```
https://godbolt.org/z/qTW1r79We

So I have the feeling the following is enough
```suggestion
  ((for_type_and_marrays<action, types, actionArgsT...>(
          std::forward<argsT>(args)..., typeList.names[typeNameIndex]),
    ++typeNameIndex), ...);
```

File: tests/sub_group_mask/sub_group_mask_extract_bits.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+using namespace sycl_cts;
+#ifdef SYCL_EXT_ONEAPI_SUB_GROUP_MASK
+
+template <typename T>
+void get_expected_bits(T &out, uint32_t mask_size, int pos) {
+  if (pos >= mask_size - 1) return;
+  int init;
+  if (pos % 2 == 0)
+    init = 1;  // 01
```
Comment (keryell) [inline tests/sub_group_mask/sub_group_mask_extract_bits.cpp:23]:
```suggestion
    init = 0b01;
```

File: tests/sub_group_mask/sub_group_mask_extract_bits.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  else
+    init = 2;  // 10
```
Comment (keryell) [inline tests/sub_group_mask/sub_group_mask_extract_bits.cpp:25]:
```suggestion
    init = 0b10;
```

File: tests/sub_group_mask/sub_group_mask_extract_bits.cpp [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+#include "sub_group_mask_common.h"
+
+#define TEST_NAME sub_group_mask_extract_bits
+
+namespace TEST_NAMESPACE {
+
+using namespace sycl_cts;
+#ifdef SYCL_EXT_ONEAPI_SUB_GROUP_MASK
+
+template <typename T>
```
Comment (keryell) [inline tests/sub_group_mask/sub_group_mask_extract_bits.cpp:18]: Add a comment explaining the magic in English

**Review by keryell — COMMENTED** [review]
Add some comments to clarify what is happening

File: tests/common/type_coverage.h (hunk 1 above)
Comment (vasilytric) [reply tests/common/type_coverage.h]: this commend was applied [here](https://github.com/KhronosGroup/SYCL-CTS/pull/202/commits/7b8b881e2665a5081119f502def79cecae04f2c3#diff-6c3f2e5414e9e68ce7030404ce12e917db441dd8be63f28613a6b951faf88c8bL270-L277)

File: tests/sub_group_mask/sub_group_mask_extract_bits.cpp (hunk 2 above)
Comment (vasilytric) [reply tests/sub_group_mask/sub_group_mask_extract_bits.cpp]: this commend was applied [here](https://github.com/KhronosGroup/SYCL-CTS/pull/202/commits/7b8b881e2665a5081119f502def79cecae04f2c3#diff-cfbc9b5df3268d3ff2e7c4f881ebbc0094b882799d25411270d0c817fc3fdd3aR23)

File: tests/sub_group_mask/sub_group_mask_extract_bits.cpp (hunk 3 above)
Comment (vasilytric) [reply tests/sub_group_mask/sub_group_mask_extract_bits.cpp]: this commend was applied [here](https://github.com/KhronosGroup/SYCL-CTS/pull/202/commits/7b8b881e2665a5081119f502def79cecae04f2c3#diff-cfbc9b5df3268d3ff2e7c4f881ebbc0094b882799d25411270d0c817fc3fdd3aR25)

File: tests/sub_group_mask/sub_group_mask_extract_bits.cpp (hunk 4 above)
Comment (kopylovanat) [reply tests/sub_group_mask/sub_group_mask_extract_bits.cpp]: Added comment here https://github.com/KhronosGroup/SYCL-CTS/pull/202/commits/9151ab676d35a569cb0c5a89104139b55590d9b5

**Review by keryell — APPROVED** [review]

**Review by keryell — APPROVED** [review]
Thanks!

**Pull request merged**